File: Coding_Practise/LeetCode/13_Roman_to_Integer/romantointeger.java
import java.util.*;

public class romantointeger{
    public static void main(String[] args){
        // System.out.println("Working");
        Scanner scanner=new Scanner(System.in);
        // String inp=scanner.next();
        String inp="MCMXCIV";
        HashMap<String,Integer> map=new HashMap<>();
        map.put("I", 1);
        map.put("V", 5);
        map.put("X", 10);
        map.put("L", 50);
        map.put("C", 100);
        map.put("D", 500);
        map.put("M", 1000);
        String[] splitData=inp.split("");
        int outi=0;
        for(int i=splitData.length-1;i>=0;i--){
            if(i<splitData.length-1 && map.get(splitData[i+1])>map.get(splitData[i])){
                outi-=map.get(splitData[i]);
            }else{
                outi+=map.get(splitData[i]);
            }
        }
        System.out.println(outi);
    }
}

File: Coding_Practise/LeetCode/13_Roman_to_Integer/romantointeger.py
inp="MCMXCIV"
dicto={
    "I":1,
    "V":5,
    "X":10,
    "L":50,
    "C":100,
    "D":500,
    "M":1000
}
inprev=inp[::-1]
outi=0
for x in range(0,len(inprev)):
    if(x>0 and dicto[inprev[x-1]]>dicto[inprev[x]]):
        outi-=dicto[inprev[x]]
    else:
        outi+=dicto[inprev[x]]
print(outi)